src/main/java/edu/harvard/iq/dataverse/DatasetPage.java

```diff
@@ -2406,8 +2407,9 @@ public String restrictSelectedFiles(boolean restricted) throws CommandException{
         } else {
             boolean validSelection = true;
             for (FileMetadata fmd : selectedFiles) {
-                if ((fmd.isRestricted() && restricted) || (!fmd.isRestricted() && !restricted)) {
+                if (fmd.isRestricted() == restricted) {
```

Comment:
simpler logic, same result - files shouldn't already be in the state that this method is changing them to.